app/portainer/views/settings/authentication/settingsAuthenticationController.js

```diff
@@ -71,6 +71,12 @@ function SettingsAuthenticationController($q, $scope, $state, Notifications, Set
     $scope.settings.AuthenticationMethod = value;
   };
 
+  $scope.onChangePasswordLength = function onChangePasswordLength(value) {
```

Comment:
still, I think you should move that state for `InternalAuth` component to it (I.e load settings and save them)

Reply:
I could do that, but why not leave it as it its since the parent already gets the settings? Is there a benefit to fetching inside a child component instead of trusting the parents data?

Reply:
less relying on angular code. and since `useQuery` relies on cache, it's not a problem to load it wherever you want

Reply:
you don't need things like the change to `app/portainer/models/settings.js` since we have typescript for that

Reply:
form validation is easier with Formik